header/store/store.go

```diff
@@ -63,49 +50,66 @@ type store struct {
 	writeHead atomic.Pointer[header.ExtendedHeader]
 	// pending keeps headers pending to be written in one batch
 	pending *batch
+
+	Params *Parameters
 }
 
 // NewStore constructs a Store over datastore.
 // The datastore must have a head there otherwise Start will error.
 // For first initialization of Store use NewStoreWithHead.
-func NewStore(ds datastore.Batching) (header.Store, error) {
-	return newStore(ds)
+func NewStore(ds datastore.Batching, opts ...Option) (header.Store, error) {
```

Comment:
We should return the concrete type now and have a proxy constructor in the FX that returns the Store instead.